process_report/tests/e2e/test_e2e_pipeline.py

```diff
@@ -0,0 +1,241 @@
+
+    return test_files
+
+
+def _prepare_pipeline_execution(
+    test_files: Dict[str, Path], workspace: Path, project_root: Path
+):
+    """Build command and environment for pipeline execution.
+
+    Args:
+        test_files: Dictionary mapping test file names to their absolute paths.
+        workspace: Path to the temporary workspace directory.
+        project_root: Path to the project root directory.
+
+    Returns:
+        Tuple[List[str], Dict[str, str]]: A tuple containing the command list and
+            environment dictionary for running the pipeline.
+    """
+    # Build command
+    command = [
+        "python",
+        "-m",
+        "process_report.process_report",
+        "--invoice-month",
+        INVOICE_MONTH,
+        "--pi-file",
+        str(test_files["test_pi.txt"]),
+        "--projects-file",
+        str(test_files["test_projects.txt"]),
+        "--timed-projects-file",
+        str(test_files["test_timed_projects.txt"]),
+        "--BU-subsidy-amount",
+        "100",
+        "--old-pi-file",
+        str(test_files["test_PI.csv"]),
+        "--alias-file",
+        str(test_files["test_alias.csv"]),
+        "--prepay-debits",
+        str(test_files["test_prepay_debits.csv"]),
+        "--prepay-credits",
+        str(test_files["test_prepay_credits.csv"]),
+        "--prepay-projects",
+        str(test_files["test_prepay_projects.csv"]),
+        "--coldfront-data-file",
+        str(test_files["test_coldfront_api_data.json"]),
+        str(test_files["test_nerc-ocp-test 2025-04.csv"]),
+        str(test_files["test_NERC OpenShift 2025-04.csv"]),
+        "--prepay-contacts",
+        str(test_files["test_prepay_contacts.csv"]),
+    ]
+
+    # Environment setup for subprocess execution
+    env = os.environ.copy()
+    # Fallback ensures test works even when CI environment doesn't set Chrome path
+    env.setdefault("CHROME_BIN_PATH", "/usr/bin/chromium")
+    env["PYTHONPATH"] = str(project_root) + ":" + env.get("PYTHONPATH", "")
+
+    return command, env
+
+
+def _run_pipeline(command: List[str], env: Dict[str, str], workspace: Path):
+    """Run the pipeline and return the result.
+
+    Args:
+        command: List of command arguments to execute the pipeline.
+        env: Environment variables dictionary.
+        workspace: Path to the temporary workspace directory where the pipeline will run.
+
+    Returns:
+        subprocess.CompletedProcess: The result of the pipeline execution.
+
+    Raises:
+        pytest.fail: If the pipeline execution times out.
+    """
+    logger.info(f"Running pipeline in: {workspace}")
+
+    try:
+        result = subprocess.run(
+            command,
+            env=env,
+            cwd=workspace,
+            capture_output=True,
+            text=True,
+            timeout=PIPELINE_TIMEOUT,
+        )
+
+        if result.stderr:
+            logger.warning(f"Pipeline stderr: {result.stderr}")
```

Comment:
Currently, there's some stuff that gets written to stderr but I don't know where the output of this logger is going. I did run pytest with `-s` to disable all capturing but I still did not see this specific warning or anything else that's being logged. 

But anyway, when I do run pytest with `-s` then I see this result anyway because it then prints the stderr.